libraries/ms-drivers/test/test_max6643_fan_controller.c

```diff
@@ -0,0 +1,83 @@
+#include "max6643_fan_controller.h"
+
+#include "interrupt.h"
+#include "log.h"
+#include "test_helpers.h"
+#include "unity.h"
+#include "wait.h"
+
+#define TEST_CALLBACK_EXPECTED_TIMES_CALLED 1
+#define TEST_FANFAIL_EXPECTED_CONTEXT 0xA
+#define TEST_OVERTEMP_EXPECTED_CONTEXT 0xB
+
+static uint8_t s_times_fanfail_callback_called;
+static uint8_t s_times_overtemp_callback_called;
+static uint8_t s_fanfail_callback_context;
+static uint8_t s_overtemp_callback_context;
+static const GpioAddress test_fanfail_pin = { .port = GPIO_PORT_A, .pin = 0 };
+static const GpioAddress test_overtemp_pin = { .port = GPIO_PORT_A, .pin = 1 };
+
+static void prv_test_max6643_fanfail_interrupt_callback(const GpioAddress *address, void *context) {
+  LOG_DEBUG("FANFAIL INTERRUPT CALLBACK TRIGGERED\n");
+
+  uint8_t *fanfail_context = (uint8_t *)context;
```

Comment:
Nitpick: you don't actually need the cast since it's a void pointer (also applies to the same line in the next function)